rfp-proposals/rfp-proposal-filscan.md

```diff
@@ -0,0 +1,141 @@
+ # RFP Proposal: Block Explorer
+ 
+ **Name of Project:** Filscan
+ 
+ **Link to RFP:**  *[rfp-block-explorer.md](https://github.com/filecoin-project/devgrants/blob/master/rfps/rfp-block-explorer.md)*
+ 
+ **RFP Category:**  `app-dev`
+ 
+ **Proposer:**   *[IPFS-FORCE](https://github.com/orgs/ipfs-force-community)*
+ 
+ **Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT and APACHE2 licenses?:**  `Yes`
+ 
+ # Project Description
+ 
+Hi Labs, we are IPFS-Force team from China with several years' development experience in Blockchain Industry and deep understanding of Blockchain Activities and Actors. With solid experience, we are capable to dive and analyze multidimensional data from blockchain raw data and displayed in creative visualization.
+ 
+FilScan is a customized blockchain engine for Filecoin with searchable and analyzed data that allows all the actors to explore the valuable information in a fast and easy way. It could be accessible by standard web browsers and responsive on mobile from anywhere in the world.
+ 
+The project will include the following sections: 
+ 
+ 1. Dashboard: Real-time Blockchain summary statistics, Search Bar(Keywords: Best Block, Peer ID and Wallet Address), Navigation Bar, Top 10 Tipset and Top-10 Message.
+ 
+ 2. Blockchain Page: Real-time and detailed Blockchain data with browse and retrieval function.
+ 
+ 3. Mining Page: Mining information which includes Miner State, Miner Index and Rank.
+ 
+ ![homepage design](https://github.com/ipfs-force-community/devgrants/blob/master/rfp-proposals/home.jpeg)
+ 
+ ## Deliverables
+ 
+ 1. The fully functional website is accessible via a https:// site with high-availability and works on mobile friendly. 
+ 
+ 2. The codebase is open-sourced and dual-licensed under MIT and APACHE2 licenses.
+ 
+ 3. Well-documented Deployment Document, API Instruction and User-Guide.
+ 
+ ## Development Roadmap
+ 
+ ### Milestone 1
+ 
+ 1. UI/UX Design: Dashboard, Blockchain Page and Mining Page
+ 
+ 2. Solution Design: Interface, Database, System Structure
```

Comment:
Can you describe how you're currently thinking about designing the backend? How will you get the data to display on the block explorer, or is it all through the chain information?

Reply:
We got all the original data from the chain. We save the result of data analysis, data statistics in database, and use Redis to make sure user can visit the web well.